CMakeLists.txt

```diff
@@ -164,7 +162,7 @@ endif()
 cmake_push_check_state()
 
 set(CMAKE_REQUIRED_INCLUDES ${CMAKE_INCLUDE_PATH} ${Boost_INCLUDE_DIRS} ${ICUE_INCLUDE_DIRS})
-set(CMAKE_REQUIRED_LIBRARIES ${Boost_LIBRARIES} ${ICU_LIBRARIES} ${PROFILE_LIBS})
+set(CMAKE_REQUIRED_LIBRARIES Boost::regex ${ICU_LIBRARIES} ${PROFILE_LIBS})
```

Comment:
I'm not exactly sure why the `check_cxx_source_runs` check following this line fails when `${Boost_LIBRARIES}` includes boost_python, yet it does. Given that the `check_cxx_source_runs` check only includes the `<boost/regex/icu.hpp>` header, it seems reasonable to only require the boost regex library.